src/RegularizedMoments.cc

```diff
@@ -0,0 +1,689 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2018  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <cmath>
+#include <exception>
+#include <string>
+#include <tuple>
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/meas/modelfit/RegularizedMoments.h"
+
+namespace lsst {
+namespace meas {
+namespace modelfit {
+
+namespace {
+/*
+ *  Anonymous namespace to contain code that calculates all the intermediate products used in
+ *  calculating the biased waited moments model value and derivative. See corresponding document
+ *  for explanation of parameter names.
+ */
+
+using Moments = MomentsModel::Moments; 
+using FirstMoment = MomentsModel::FirstMoment;
+using SecondMoment = MomentsModel::SecondMoment;
+
+// Builds moments with specific values used in the tests for the anonymous namesapce functionality
+std::tuple<Moments, Moments> buildTestMoments(){
+    Moments Q, W;
+    Q << 6, 4, 3, 2, 1, 4;
+    W << 2, 4, 3.1, 2.5, 1.2, 3.7;
+    return std::make_tuple(Q, W);
+}
+
+struct AlphaX {
```

Comment:
Some documentation for `AlphaX` and its members would be nice ;)

Reply:
added